pkg/controller/registry/resolver/resolver.go

```diff
@@ -302,25 +309,27 @@ func (resolver *MultiSourceResolver) resolveCRDDescription(sources map[registry.
 	}
 
 	// registry.SourceKey for source found to contain the CRD
-	crdSrcKey := firstSrcKey
+	crdSourceKey := preferredSourceKey
 
-	// Attempt to get the first source
-	source, ok := sources[firstSrcKey]
+	// Attempt to get the preferredSource
+	source, ok := sources[preferredSourceKey]
```

Comment:
There's a ton of code scattered throughout for managing the "preferred source". Ideally, these methods should not have to care about which source is preferred. This isn't something we have to fix now, it can be a followup or it can wait until we have better plans for a total-ordering of sources.

Reply:
We could always change the ResolveInstallPlan sources parameter to be a pre-ordered, pre-filtered slice of Sources. Another possible alternative is that we establish some criteria for general Source priority and pass in a priority queue.